src/Dialect/ONNX/ONNXOps.td.inc

```diff
@@ -4387,6 +4387,7 @@ def ONNXMaxOp:ONNX_Op<"Max",
 
 def ONNXMaxPoolOp:ONNX_Op<"MaxPool",
   [Pure, DeclareOpInterfaceMethods<ShapeInferenceOpInterface>, DeclareOpInterfaceMethods<ShapeHelperOpInterface>]> {
+  let hasCanonicalizer = 1;
```

Comment:
Sorry for the confusion. I think this is not MaxPool you are dealing with, it should be `ONNXMaxPoolSingleOutOp` in https://github.com/onnx/onnx-mlir/blob/main/src/Dialect/ONNX/AdditionalONNXOps.td#L429. `gen_onnx_mlir.py` does not update `AdditionalONNXOps.td`. So please add `hasCanonicalizer    = 1` to`ONNXMaxPoolSingleOutOp` in `AdditionalONNXOps.td` instead.